### DISCUSSION TITLE
Rule without active triggers will act on all transactions

### DISCUSSION THREAD

**Discussion by jafin** [firefly-iii/firefly-iii#5726] [Firefly III]
Version: 5.6.14
Runtime: Docker

I've been adding rules via the API.
I used the data importer to import a credit card statement

An unexpected outcome occurred, one rule appears to apply to almost all imported transactions (where no other rule matched). And I'm uncertain why.

The rule trigger is "Description contains 'value'"   Note: Value does start with an Asterix, but I have other triggers with similar pattern and they do not exhibit the same behaviour.
And all it does is set the `Category` and `Destination account` for the entry.

below is a query from the database.  The thing that stood out is for this rule the `active` flag is `false` for the trigger.  All other rule triggers are `active`.  
This is possibly a side effect of when I was creating rules via the API and inadvertently set this rule to `active:false` on creation.

![image](https://user-images.githubusercontent.com/127927/152707445-3d347d15-8d74-4f43-a984-a62c981f2577.png)

When I use the UI to see `matching transactions`, only 2 transactions are found that match the description (expected).
I am wondering if that fact that the rule trigger is `active:false`, and therefore that may get seen as no active triggers, would this cause it to match against everything?

I'll do some more investigation again tonight, I'll update the active state on the `description_contains` to `true` to see if that is the cause.

**Comment by JC5** [comment]
I've edited the title just in case, but it feels that this could be the case. I'll check it out and add a validation rule if necessary.

**Reply by jafin** [reply]
Here are the results of some tests.

Confirming that there is some sort of error/bug scenario here when the (only?) trigger is `active:false`:

* If the import is run, with a rule that has the only trigger with state `active:false` then it appears the rule fires for all transactions (possibly unmatched by other rules) (not expected)
* If the import is run, with a rule that has the trigger with state `active:true`  then it acts as expected.

When using the UI if I attempt to remove the `active:false` trigger from the UI, by clicking the trashbin, and click Update rule, then the UI shows the rule trigger is gone, but then if I refresh the page, the rule trigger still exists.

**Reply by JC5** [reply]
Thanks, I can confirm this. The UI won't be updated but I'm making sure that you won't be able to submit rules that only have inactive rules/triggers.

**Reply by JC5** [reply]
This should work as expected on the demo site, check it out.
